tmt/steps/execute/__init__.py

```diff
@@ -604,8 +604,21 @@ def wake(self) -> None:
 
     def summary(self) -> None:
         """ Give a concise summary of the execution """
-        tests = fmf.utils.listed(self.results(), 'test')
-        self.info('summary', f'{tests} executed', 'green', shift=1)
+        results = list(self.results())
+        tests = fmf.utils.listed(
+            filter(
+                lambda r: r.result != ResultOutcome.NOT_EXECUTED,
+                results),
+            'test')
+
+        def filter_function(result: tmt.result.Result) -> bool:
+            return bool(result.result == ResultOutcome.NOT_EXECUTED)
+        if next(filter(filter_function, results), None) is not None:
+            not_executed = ', ' + \
+                fmf.utils.listed(filter(filter_function, results), 'test') + ' not executed'
+        else:
+            not_executed = ''
+        self.info('summary', f'{tests} executed' + not_executed, 'green', shift=1)
```

Comment:
A couple of issues, mostly WRT how the message is put together - I especially despise the comma hidden between lines :)

How about something with `join()`? `self.results()` is already a list, BTW.

```python
executed_tests = filter(lambda r: r.result != ResultOutcome.NOT_EXECUTED, self.results())
skipped_tests = filter(lambda r: r.result == ResultOutcome.NOT_EXECUTED, self.results())

message = [
    fmf.utils.listed(executed_tests, 'test') + ' executed'
]

if skipped_tests:
   message.append(fmf.utils.listed(skipped_tests, 'test') + ' not executed')

self.info('summary', ', '.join(message), 'green', shift=1)
```